tests/Bridge/Symfony/Validator/Metadata/Property/ValidatorPropertyMetadataFactoryTest.php

```diff
@@ -186,11 +219,117 @@ public function testCreateWithPropertyWithValidationConstraints()
+        $decoratedPropertyMetadataFactory->create(DummyValidatedEntity::class, $property, [])->willReturn(
+                new PropertyMetadata(new Type(Type::BUILTIN_TYPE_STRING))
+            )->shouldBeCalled();
+
+        $lengthRestrictions = new PropertySchemaLengthRestriction();
+        $validatorPropertyMetadataFactory = new ValidatorPropertyMetadataFactory(
+            $validatorMetadataFactory->reveal(), $decoratedPropertyMetadataFactory->reveal(), [$lengthRestrictions]
+        );
+
+        $schema = $validatorPropertyMetadataFactory->create(DummyValidatedEntity::class, $property)->getSchema();
+        $this->assertNotNull($schema);
+        $this->assertArrayHasKey('minLength', $schema);
+        $this->assertArrayHasKey('maxLength', $schema);
+
+        $numberTypes = [Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT];
+
+        foreach ($numberTypes as $type) {
+            $decoratedPropertyMetadataFactory->create(DummyValidatedEntity::class, $property, [])->willReturn(
+                    new PropertyMetadata(new Type($type))
+                )->shouldBeCalled();
+            $validatorPropertyMetadataFactory = new ValidatorPropertyMetadataFactory(
+                $validatorMetadataFactory->reveal(), $decoratedPropertyMetadataFactory->reveal(), [$lengthRestrictions]
+            );
+
+            $schema = $validatorPropertyMetadataFactory->create(DummyValidatedEntity::class, $property)->getSchema();
+            $this->assertNotNull($schema);
+            $this->assertArrayHasKey('minimum', $schema);
+            $this->assertArrayHasKey('maximum', $schema);
+        }
+    }
+
+    public function testCreateWithPropertyRegexRestriction(): void
+    {
+        $validatorClassMetadata = new ClassMetadata(DummyValidatedEntity::class);
+        (new AnnotationLoader(new AnnotationReader()))->loadClassMetadata($validatorClassMetadata);
+
+        $validatorMetadataFactory = $this->prophesize(MetadataFactoryInterface::class);
+        $validatorMetadataFactory->getMetadataFor(DummyValidatedEntity::class)
+                                 ->willReturn($validatorClassMetadata)
+                                 ->shouldBeCalled();
+
+        $decoratedPropertyMetadataFactory = $this->prophesize(PropertyMetadataFactoryInterface::class);
+        $decoratedPropertyMetadataFactory->create(DummyValidatedEntity::class, 'dummy', [])->willReturn(
+                new PropertyMetadata()
+            )->shouldBeCalled();
+
+        $validationPropertyMetadataFactory = new ValidatorPropertyMetadataFactory(
+            $validatorMetadataFactory->reveal(), $decoratedPropertyMetadataFactory->reveal(),
+            [new PropertySchemaRegexRestriction()]
+        );
+
+        $schema = $validationPropertyMetadataFactory->create(DummyValidatedEntity::class, 'dummy')->getSchema();
+        $this->assertNotNull($schema);
+        $this->assertArrayHasKey('pattern', $schema);
+        $this->assertSame('^dummy$', $schema['pattern']);
```

Comment:
Wouldn't the test be more descriptive with something like `assertEquals($schema, ['pattern' => '^dummy$'])`?

Reply:
fixed